crates/shared/src/price_estimation/trade_finder.rs

```diff
@@ -425,11 +425,11 @@ mod tests {
             out_amount: 2_000_000_u128.into(),
             gas_estimate: 133_700,
             approval: None,
```

Comment:
One issue I foresee is that multiple approvals may be needed for each interaction. I think I'm OK with having it like this for now (as in theory, the `driver` would encode the `approval` as an interaction) **but**, in that case, I think that the JSON model that we parse from the driver should disallow approvals being sent.